app/main/posts/views/share/post-export.directive.js

```diff
@@ -28,54 +29,85 @@ function PostExportController(
     ConfigEndpoint,
     Notify,
     $q,
-    PostFilters
+    PostFilters,
+    _
 ) {
     $scope.loading = false;
-    $scope.exportPosts = exportPosts;
-
-    function exportPosts() {
+    this.getQuery = getQuery;
+    this.prepareExport = prepareExport;
+    this.showCSVResults = showCSVResults;
+    this.requestExport = requestExport;
+    $scope.exportPostsConfirmation = function () {
+        /**
+         * Trigger confirm notification for user.
+         * When the user accepts, get the CSV
+         */
         Notify.confirm('notify.post.export').then(function (message) {
+            prepareExport();
+        });
+    };
+    function prepareExport() {
+        loadingStatus(true);
+        var site = ConfigEndpoint.get({ id: 'site' }).$promise;
+        var query = getQuery();
+        var exportQuery = PostEndpoint.export(query);
+        requestExport(site, query, exportQuery);
+    }
+
+    function requestExport(site, query, exportQuery) {
```

Comment:
This is not a block to merging but it would be good to add a test covering this function.

Reply:
While I agree with the sentiment, and I'd always choose to have higher coverage, I believe with *unit tests*, all we can test here in requestExport is how a  mock of $q responds to an error or success response, since there is no logic whatsoever in the callbacks?. 
Any help with how to cover this function correctly would be appreciated. 

How I see it:
We could add tests separately for showCSVResults, as you mentioned, and for loadingStatus to check that $scope.loading is correct and that notifyProgress or notify were called, which would be more significative and give us more insights on how things are working. This is a good example of why we would want to have good integration tests in addition to unit tests. :/ .

Reply:
Agree